src/modules/ecdsa_adaptor/main_impl.h

```diff
@@ -341,6 +341,15 @@ int secp256k1_ecdsa_adaptor_recover(const secp256k1_context* ctx, unsigned char
      * branch point. */
     secp256k1_declassify(ctx, &enckey_expected_ge, sizeof(enckey_expected_ge));
     if (!secp256k1_eckey_pubkey_serialize(&enckey_expected_ge, enckey_expected33, &size, SECP256K1_EC_COMPRESSED)) {
+        /* Unreachable in tests
```

Comment:
I don't understand. Do you mean "Unreachable in VERIFY builds"? 

If yes, I think we should avoid this. In other words: If `scalar_inverse` expects valid scalars, we should always give it valid scalars.

Reply:
Ah so it's "unreachable if the user doesn't fiddle around with the data in ecdsa_signature" (but we still return 0 because it's the safest thing to do)?

I guess that's fine but maybe the comment can be improved. It was not clear to me. Maybe mention why this is unreachable in production and mention test coverage.

Reply:
Mentioned test coverage in the comment.